File: 2022/day01/part2.m
% Advent of Code 2022
% Day 1
% Part 2
clc
clear all
data = dlmread("input.txt",'\n');

numValues = max(size(data));
maxElves = 3;

zeroArray = find(~data);
numElves = length(zeroArray);
calorieArray = zeros(size(zeroArray));

for i = 1:numElves
    if i == 1
        calorieArray(i) = sum(data(1,1:zeroArray(i)));
    else
        calorieArray(i) = sum(data(1,zeroArray(i-1):zeroArray(i)));
    end
end

maximumCalories = maxk(calorieArray,maxElves);
maximumCaloriesSum = sum(maximumCalories);

fprintf('The three elves with the most calories had %d Calories in total!\n',maximumCaloriesSum)

File: 2022/day01/part2.py
# Day 1 of AOC 2022
import numpy as np

with open("day01/input.txt") as f:
    data = f.read().splitlines()
    
maxelf = np.zeros([3,], dtype = int)
elf = 0
for snack in data:
    if snack != "":
        elf += int(snack)
    else:
        if elf > np.min(maxelf):
            maxelf[maxelf.argmin()] = int(elf)
        elf = 0
            

print(f"The three top elves are holding: {np.sum(maxelf)} Calories!")
